llvm-external-projects/iree-compiler-api/lib/CAPI/Compiler.cpp

```diff
@@ -59,7 +64,28 @@ struct CompilerOptions {
 
 DEFINE_C_API_PTR_METHODS(IreeCompilerOptions, CompilerOptions)
 
-void ireeCompilerRegisterTargetBackends() { registerHALTargetBackends(); }
+void ireeCompilerRegisterAllDialects(MlirContext context) {
+  DialectRegistry registry;
+  registerAllDialects(registry);
+  unwrap(context)->appendDialectRegistry(registry);
+}
+
+void ireeCompilerRegisterAllPasses() {
+  registerAllPasses();
+}
+
+void ireeCompilerRegisterAllIREETranslations() {
```

Comment:
I expect that these are actually not required. Could you confirm that you need them? Some of them come with a significant binary size impact and are not used.

Reply:
I confirm that it's not required for my test case. I just placed there trying to follow what `iree-translate` code did on startup while trying to get things to work, I didn't notice it could be removed later.